<launch>
  <rosparam ns="client_params">
    vel_topic: 'navigation_velocity_smoother/raw_cmd_vel'
  </rosparam>

  <node pkg="rosduct" name="rosduct" type="rosduct_main.py" output="screen" respawn="true">
    <rosparam>
    # ROSbridge websocket server info
    rosbridge_ip: localhost
    rosbridge_port: 9090
    # Topics being published remotely to expose locally
    remote_topics: [
                        ['/turtlebot_agent/active/goal', 'long_term_deployment/TaskActionGoal', '/robot_client/active/goal'],
                        ['/turtlebot_agent/active/cancel', 'actionlib_msgs/GoalID', '/robot_client/active/cancel'],
                        ['/turtlebot_agent/continuous/goal', 'long_term_deployment/TaskActionGoal', '/robot_client/continuous/goal'],
                        ['/turtlebot_agent/continuous/cancel', 'actionlib_msgs/GoalID', '/robot_client/continuous/cancel']
                   ]
    # Topics being published in the local roscore to expose remotely
    local_topics: [
                        ['/robot_client/active/result', 'long_term_deployment/TaskActionResult', '/turtlebot_agent/active/result'],
                        ['/robot_client/active/feedback', 'long_term_deployment/TaskActionFeedback', '/turtlebot_agent/active/feedback'],
                        ['/robot_client/active/status', 'actionlib_msgs/GoalStatusArray', '/turtlebot_agent/active/status'],
                        ['/robot_client/continuous/result', 'long_term_deployment/TaskActionResult', '/turtlebot_agent/continuous/result'],
                        ['/robot_client/continuous/feedback', 'long_term_deployment/TaskActionFeedback', '/turtlebot_agent/continuous/feedback'],
                        ['/robot_client/continuous/status', 'actionlib_msgs/GoalStatusArray', '/turtlebot_agent/continuous/status']
                  ]
    # Services running remotely to expose locally
    remote_services: [
                        ['/task_server/register_agent', 'long_term_deployment/RegisterAgent'],
                        ['/task_server/unregister_agent', 'long_term_deployment/UnregisterAgent'],
                        ['/task_server/get_agents', 'long_term_deployment/GetRegisteredAgents']
                     ]
    # Services running locally to expose remotely
    local_services: [
                        ['/robot_client/active/get_goal_from_id', 'long_term_deployment/GetTaskFromID', '/turtlebot_agent/active/get_goal_from_id'],
                        ['/robot_client/continuous/get_goal_from_id', 'long_term_deployment/GetTaskFromID', '/turtlebot_agent/continuous/get_goal_from_id']
                    ]
    # Parameters to be sync, they will be polled to stay in sync
    #parameters: []
    #parameter_polling_hz: 1

    </rosparam>
  </node>
  <node pkg="long_term_deployment" name="robot_client" type="client.py" output="screen">
    <param name="agent_name" value="turtlebot"/>
  </node>
</launch>
